st3/sublime_lib/resource_path.py

```diff
@@ -11,7 +10,35 @@
 __all__ = ['ResourcePath']
 
 
-@total_ordering
+ROOT_ORDER = {
+    'Packages': -2,
+    'Cache': -1,
+}
+
+
+def sort_by_load_order(paths):
+    default_packages = Path(sublime.executable_path()).parent.glob('*.sublime-package')
+    installed_packages = Path(sublime.installed_packages_path()).glob('*.sublime-package')
+
+    def sort_key(path):
+        package = path.package
+
+        if package == 'Default':
+            package_order = -3
+        elif package == 'User':
+            package_order = 1
+        elif package in installed_packages:
+            package_order = -1
+        elif package in default_packages:
+            package_order = -2
+        else:
+            package_order = 0
+
+        return (ROOT_ORDER[path.root], path.root, package_order) + path.parts[1:]
```

Comment:
```suggestion
        return (ROOT_ORDER[path.root], package_order) + path.parts[1:]
```
Like this, right?

Reply:
That wouldn't sort nonstandard roots (e.g. `A/Foo` before `Z/Foo`). I'm not sure whether the order really matters, but even for the current limited use I think it's best to do a total sort.

Reply:
Right, duh. Fixed.